src/js/gi/reducers/calculator.js

```diff
@@ -45,8 +45,8 @@ export default function (state = INITIAL_STATE, action) {
       ].includes(field);
 
       if (isDollarAmount && !isFinite(value)) {
-        const dollarAmount = value[0] === '$' ? +value.substring(1) : +value;
-        convertedValue = isFinite(dollarAmount) ? dollarAmount : '';
+        // Strip all non-numeric characters.
+        convertedValue = +value.replace(/[^0-9\.]+/g, '');
```

Comment:
Yeah, it might be preferred to keep decimals. The calculations end up getting rounded anyway, so I'm not sure it's necessary. But I'm not sure of the best way to enforce the displayed input to be formatted with dollar signs and commas without converting them back to regular numbers in calculations.